File: go/emit_log.go
package main

import (
	"github.com/streadway/amqp"
	"log"
	"os"
	"fmt"
)

func failOnError(err error, msg string) {
	if err != nil {
		log.Fatalf("%s: %s", msg, err)
		panic(fmt.Sprintf("%s: %s", msg, err))
	}
}

func main() {
	conn, err := amqp.Dial("amqp://guest:guest@localhost:5672/")
	failOnError(err, "Failed to connect to RabbitMQ")
	defer conn.Close()

	ch, err := conn.Channel()
	failOnError(err, "Failed to open a channel")
	defer ch.Close()

	err = ch.ExchangeDeclare(
		"logs",    // name
		"fanout",  // type
		true,      // durable
		false,     // auto-deleted
		false,     // internal
		false,     // noWait
		nil,       // arguments
	)
	failOnError(err, "Failed to declare an exchange")

	body := bodyFrom(os.Args)
	err = ch.Publish(
		"logs", // exchange
		"",     // routing key
		false,  // mandatory
		false,  // immediate
		amqp.Publishing{
			ContentType:     "text/plain",
			Body:            []byte(body),
		})

	failOnError(err, "Failed to publish a message")
	log.Printf(" [x] Sent %s", body)

	os.Exit(0)
}

func bodyFrom(args []string) string {
	var body string
	if (len(args) < 1) || os.Args[1] == "" {
		body = "hello"
	} else {
		body = os.Args[1]

	}

	return body
}

File: dart/emit_log.dart
import "package:dart_amqp/dart_amqp.dart";

void main (List<String> arguments) {
    ConnectionSettings settings = new ConnectionSettings(
        host: "localhost"
    );

    Client client = new Client(settings: settings);

    String msg = arguments.isEmpty ? "Hello World!" : arguments[0];

    client
        .channel()
        .then((Channel channel) =>
                channel.exchange("logs", ExchangeType.FANOUT, durable: false))
        .then((Exchange exchange) {
            exchange.publish(msg, null);
            print(" [x] Sent ${msg}");
            return client.close();
        });
}
